util/src/main/java/com/coderanch/util/cli/InputUtility.java

```diff
@@ -0,0 +1,230 @@
+     */
+    public double nextDouble(String prompt, Predicate<Double> doublePredicate) throws IOException {
+        System.out.println(prompt);
+        try(var br = new BufferedReader(new InputStreamReader(inputStream))){
+            while(true) {
+              var line = br.readLine();
+                try {
+                    var num = Double.parseDouble(line);
+                    if(doublePredicate.test(num)) {
+                        return num;
+                    }
+                    else {
+                        System.out.println("Invalid input.");
+                    }
+                } catch (Exception e) {
+                    System.out.println("Must be double.");
+                }
+            }
+        }
+    }
+
+    /**
+     * Calls {@link InputUtility#nextString(String, Predicate)}.
+     * Passes the prompt and a predicate that validates "y", "n", "yes", "no".
+     * 
+     * @param prompt the prompt to display to the user.
+     * @param stringPredicate the predicate to use for validation.
+     * @return true for yes and false for no.
+     * @throws IOException
+     */
+    public boolean nextYesNo(String prompt, Predicate<String> stringPredicate) throws IOException {
+        var result = this.nextString(prompt, yesOrNo()).toLowerCase();
+        if(YES.contains(result)) {
+            return true;
+        }
+        else return false;
+    }
+
+    /**
+     * Calls {@link InputUtility#pause(String)}.
+     * The prompt defaults to "Press when ready".
+     *
+     * @throws IOException
+     */
+    public void pause() throws IOException {
+        pause("Press when ready");
+    }
+
+    /**
+     * Calls nextString{@link InputUtility#nextString(String, Predicate)}.
+     * Uses prompt entered and waits for the user to press ".".
+     *
+     * @param prompt the prompt to display to the user.
+     * @throws IOException
+     */
+    public void pause(String prompt) throws IOException {
+        this.nextString(prompt, s -> s.trim().equals("."));
+    }
+
+    /**
+     * Get a String predicate that tests for (case insensitive) "y", "n", "yes", "no".
+     *
+     * @return  a String predicate that tests for (case insensitive) "y", "n", "yes", "no".
+     */
+    public static Predicate<String> yesOrNo() {
+        return s -> {
+            var cleanedString = s.trim().toLowerCase();
+            if(YES.contains(s)) {
+                return true;
+            }
+            else if(NO.contains(s)) {
+                return true;
+            }
+            return false;
+        };
+    }
+
+    /**
+     * Get a String predicate that takes two or more Strings and tests whether any one of them match the input.
+     *
+     * @param these the strings for the input to match to.
+     * @return a String predicate that takes two or more Strings and tests whether any one of them match the input.
+     */
+    public static Predicate<String> oneOfThese(String...these){
+        return s -> {
```

Comment:
Before this, what about a test to see if more than one String was passed?  Maybe throw an IllegalArgumentException if not.